tests/perf-system/analyzer/analyzer.py

```diff
@@ -130,6 +130,57 @@ def customize_table(self, fields_list: List[str], values_list: List[List]):
             custom_table.add_row(val_row)
         return custom_table
 
+    def plot_commits(self, df_responses: pd.DataFrame, df_generator: pd.DataFrame):
+        """
+        For submitted requests of the type: N posts 1 commit, this function will
+        plot the number of posted versus committed messages
+        """
+        init_post_id = 0
+        init_time = 0
+        custom_header = "x-ms-ccf-transaction-id"
+        for head in df_responses.iloc[0]["rawResponse"].split("\n"):
+            if custom_header in head.split(":")[0]:
+                init_post_id = int(head.split(":")[1].split(".")[1][:-1]) - 1
+                init_time = float(df_responses.iloc[0]["receiveTime"])
+        post_ids = []
+        committed_ids = []
+        time_units = []
+        for row in range(len(df_generator.index)):
+            if df_generator.iloc[row]["request"].split(" ")[
+                0
+            ] == "GET" and df_generator.iloc[row]["request"].split(" ")[1].endswith(
+                "commit"
+            ):
+                # Break when the first of the aggressive commits
+                # is equal to the posts
+                if (
+                    len(post_ids) > 1
+                    and post_ids[-1] == post_ids[-2]
+                    and post_ids[-1] == committed_ids[-1]
+                ):
+                    break
+
+                res_headers = df_responses.iloc[row]["rawResponse"].split("\n")
+                for h in res_headers:
+                    if custom_header in h.split(":")[0]:
+                        # get the latest post_id from the committed message

```

Comment:
That seems extremely specific to one kind of request/generated payloads. It would be good to be explicit about this, and perhaps make it pluggable.